packages/ui-react/src/components/PaymentForm/PaymentForm.tsx

```diff
@@ -1,121 +1,108 @@
+            const issuer = Payment.fns.cardType(context.parent.cardNumber);
 
-      const referrer = window.location.href;
-      const returnUrl = createURL(window.location.href, { u: 'waiting-for-payment' });
-
-      await checkoutController.directPostCardPayment({ couponCode, ...paymentData.values }, referrer, returnUrl);
-
-      intervalCheckAccess({
-        interval: 15000,
-        callback: (hasAccess) => hasAccess && navigate(modalURLFromLocation(location, 'welcome')),
-      });
-    },
-    object().shape({
-      cardNumber: string().test('card number validation', t('checkout.invalid_card_number'), (value) => {
-        return Payment.fns.validateCardNumber(value as string);
-      }),
-      cardExpiry: string().test('card expiry validation', t('checkout.invalid_card_expiry'), (value) => {
-        return Payment.fns.validateCardExpiry(value as string);
+            return Payment.fns.validateCardCVC(value as string, issuer);
+          }),
+        cardExpMonth: string().required(),
+        cardExpYear: string().required(),
       }),
-      cardCVC: string().test('cvc validation', t('checkout.invalid_cvc_number'), (value) => {
-        const issuer = Payment.fns.cardType(paymentData?.values?.cardNumber);
-        return Payment.fns.validateCardCVC(value as string, issuer);
-      }),
-    }),
-    true,
-  );
+    validateOnBlur: true,
+    onSubmit: onPaymentFormSubmit,
+  });
 
   useEffect(() => {
-    if (paymentData.values.cardExpiry) {
-      const expiry = Payment.fns.cardExpiryVal(paymentData.values.cardExpiry);
+    if (values.cardExpiry) {
+      const expiry = Payment.fns.cardExpiryVal(values.cardExpiry);
       if (expiry.month) {
-        paymentData.setValue('cardExpMonth', expiry.month.toString());
+        setValue('cardExpMonth', expiry.month.toString());
       }
       if (expiry.year) {
-        paymentData.setValue('cardExpYear', expiry.year.toString());
+        setValue('cardExpYear', expiry.year.toString());
       }
     }
     //eslint-disable-next-line
-  }, [paymentData.values.cardExpiry]);
+  }, [values.cardExpiry]);
 
   return (
     <div className={styles.paymentForm}>
-      <div>
-        <TextField
-          label={t('checkout.card_holder_name')}
-          name="cardholderName"
-          value={paymentData?.values?.cardholderName}
-          onChange={paymentData?.handleChange}
-          onBlur={paymentData?.handleBlur}
-          placeholder={t('checkout.credit_card_name')}
-          required
-        />
-      </div>
-      <div>
-        <CreditCardNumberField
-          value={paymentData?.values?.cardNumber?.toString()}
-          error={paymentData?.errors?.cardNumber}
-          onBlur={paymentData?.handleBlur}
-          onChange={paymentData?.handleChange}
-        />
-      </div>
-      <div className={styles.columns}>
+      <form onSubmit={handleSubmit} data-testid={testId('payment_form')} noValidate>
```

Comment:
I'm a bit suprised to see this change in this PR. Can you elaborate?

Reply:
This component was using useForm, but not a <form/> element with a submit handler. Also the error wasn't presented anywhere.